### PR TITLE
[DOCS] Adds Features subsection to Overview

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a `Features` subsection to the `Overview` that contains a bulleted list of client features.
It also moves the compatibility matrix from `Overview` to `Installation`.

### Preview

[GO Overview](https://go-elasticsearch_195.docs-preview.app.elstc.co/guide/en/elasticsearch/client/go-api/master/overview.html)

### CODE REVIEW

File: .doc/installation.asciidoc [hunk 1]
```diff
@@ -48,3 +48,32 @@ END
...
 ------------------------------------
+
+
+[discrete]
+=== {es} Version Compatibility
+
+The client major versions correspond to the {es} major versions: to connect to 
+{es} `7.x`, use a `7.x` version of the client, to connect to {es} `6.x`, use a 
+`6.x` version of the client, and so on.
+
```
Comment (philkra) [inline .doc/installation.asciidoc:59]: Can we add wording, that while a `5.x` client is available, it's no longer actively maintained given that neither is any 5.x version of Elasticsearch. Could link to the EOL policy: https://www.elastic.co/support/eol

**Review by philkra — CHANGES_REQUESTED** [review]

File: .doc/installation.asciidoc (hunk 1 above)
Comment (szabosteve) [reply .doc/installation.asciidoc:59]: Thanks @philkra! I've added a note via 2bdf729.

**Review by philkra — APPROVED** [review]
LGTM, thanks a lot @szabosteve

**Comment by szabosteve** [comment]
**Backport commit:**
7.x: 152864b

**Comment by tty2** [comment]
@szabosteve @philkra have you seen that with this PR actions (integration tests) failed?
I'm curious how did you merge it? Is there an option to merge PR if it didn't pass tests?

**Pull request merged**